### ISSUE TITLE
Question regarding details on Table 1

### ISSUE THREAD

**Issue by amy-hyunji** [NVlabs/hymba#10]
Hi, Thanks for releasing a great work.

I have a few questions regarding details on training Table 1 Row A and B in the paper, where you compare fusion using sequential and parallel formats:
1. Are the attention layers for row A and B all full attentions without the use of sliding window attention? If so, have you also tried with replacing them with sliding window attention and could still see the benefits?
2. Were they trained following the same produres as in Figure 8 (generation pretraining -> LR annealing) with identical hyperparameters? Also, what dataset was used for training?
3. Did the parallel format also show lower validation loss compared to the sequential format?

Also, I wonder why you averaged the mamba output and SWA output with hidden states dimension multiplied by two rather than after projecting the mamba output and averaging them with SWA output

I would appreciate any clarification you can provide. Thank you!

**Comment by XinDongol** [comment]
1. Row A and B are all full Attentions and Mamba. Yes, we tried SWA and Mamba in parallel, where the conclusion still holds. Please check the ICLR discussion. 
2. For Figure1, since it is small scale, we use cosine lr (5e-4) scheduler on FineWeb-Edu 100B tokens. Note that apple-to-apple comparisons are done under exactly the same hyperparameters across different architectures. 
3. Yes.
4. You can do Mamba projecting and then average. In our design, we hope the Mamba and Attention heads can share as much as possible. So we average them before output projection. 
Hope these answer all your questions.

**Comment by amy-hyunji** [comment]
Thanks for the quick response! It answered many of my questions. 

I’m currently working on replicating the configurations in Table 1 (Rows A & B), but I’ve encountered some issues and would like to pinpoint where the difference might be. I have a few more questions:

1. For exp. in rows A and B, like those in Hymba block implementation, there are no MLP layers or residual connections, right?
2. Have you tried using the Mamba projection and then averaging the outputs? If so, did you observe the same trend? 

Also, regarding Figure 7 in the paper, which layer was used for the analysis shown there?

**Comment by XinDongol** [comment]
1. There are MLP and residual conncetions. They are all complete model. 
2. It does not change trend.

**Comment by amy-hyunji** [comment]
Thanks for the response.

For Q1, I was mentioning in terms of the block. I assume that the model in row B would be the same block as here?: https://github.com/NVlabs/hymba/blob/df5b2a95b9905534eb162305a5414bc43e37e84b/barebones_hymba/barebones_hymba_block.py#L384?

For Q2, then would be the reason for choosing the current form (averaging them before output projection) because it showed higher performance?

**Comment by XinDongol** [comment]
Q1: yes
Q2: because we hope to reuse parameters for Mamba and Attention.
